src/main/java/org/broadinstitute/hellbender/utils/fermi/FermiLiteAssembly.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.hellbender.utils.fermi;
+
+import java.io.*;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+/** an assembly is just a collection of contigs */
+public final class FermiLiteAssembly {
+    private final List<Contig> contigs;
+
+    FermiLiteAssembly(final List<Contig> contigs ) { this.contigs = contigs; }
+
+    public int getNContigs() { return contigs.size(); }
+    public Contig getContig( final int idx ) { return contigs.get(idx); }
+    public List<Contig> getContigs() { return Collections.unmodifiableList(contigs); }
```

Comment:
Not set in the constructor.  (Can't be because the connections graph is cyclic.)  Wrapped it in setConnections method, which should address your valid concern.